src/AcmeClient.php

```diff
@@ -11,79 +11,265 @@
+            sleep(1);
+        }
+
+        if ('pending' === $response['status']) {
+            throw new HttpChallengeTimedOutException($response);
+        } elseif ('invalid' === $response['status']) {
+            throw new HttpChallengeFailedException($response);
+        }
+
+        return $response;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getCertificatesChain()
+    public function requestCertificate($domain, CertificateRequest $csr, $timeout = 180)
     {
-        return $this->certificatesChain;
+        Assert::stringNotEmpty($domain, 'requestCertificate::$domain expected a non-empty string. Got: %s');
+        Assert::integer($timeout, 'requestCertificate::$timeout expected an integer. Got: %s');
+
+        $humanText = ['-----BEGIN CERTIFICATE REQUEST-----', '-----END CERTIFICATE REQUEST-----'];
+
+        $csrContent = $this->csrSigner->signCertificateRequest($csr);
+        $csrContent = trim(str_replace($humanText, '', $csrContent));
+        $csrContent = trim($this->httpClient->getBase64Encoder()->encode(base64_decode($csrContent)));
+
+        $response = $this->requestResource('POST', ResourcesDirectory::NEW_CERTIFICATE, [
+            'resource' => ResourcesDirectory::NEW_CERTIFICATE,
+            'csr'      => $csrContent,
+        ], false);
+
+        // If the CA has not yet issued the certificate, the body of this response will be empty
+        if (strlen(trim($response)) < 10) { // 10 to avoid false results
+            $location = $this->httpClient->getLastLocation();
+
+            // Waiting loop
+            $waitingTime = 0;
+
+            while ($waitingTime < $timeout) {
+                $response = $this->httpClient->unsignedRequest('GET', $location, null, false);
+
+                if (200 === $this->httpClient->getLastCode()) {
+                    break;
+                }
+
+                if (202 !== $this->httpClient->getLastCode()) {
+                    throw new CertificateRequestFailedException($response);
+                }
+
+                $waitingTime++;
+                sleep(1);
+            }
+
+            if (202 === $this->httpClient->getLastCode()) {
+                throw new CertificateRequestTimedOutException($response);
+            }
+        }
+
+        // Find issuers certificate
+        $certificatesChain = null;
+
+        foreach ($this->httpClient->getLastLinks() as $link) {
+            if (!isset($link['rel']) || 'up' !== $link['rel']) {
+                continue;
+            }
+
+            $location = substr($link[0], 1, -1);
+            $certificate = $this->httpClient->unsignedRequest('GET', $location, null, false);
+
+            if (strlen(trim($certificate)) > 10) {
+                $pem = chunk_split(base64_encode($certificate), 64, "\n");
+                $pem = "-----BEGIN CERTIFICATE-----\n".$pem."-----END CERTIFICATE-----\n";
+
+                $certificatesChain = new Certificate($pem, $certificatesChain);
```

Comment:
I couldn't find any example where multiple certificates were chained in a single header, I'm not sure of this part indeed. I'll check more.